src/main/java/no/digipost/signature/client/core/internal/ClientHelper.java

```diff
@@ -96,30 +92,31 @@ public void cancel(Cancellable cancellable) {
             String url = cancellable.getCancellationUrl().getUrl();
             Response response = postEmptyEntity(url);
             Status status = Status.fromStatusCode(response.getStatus());
-            if (status == CONFLICT) {
+            if (status == OK) {
+                return;
+            } else if (status == CONFLICT) {
                 throw new NotCancellableException();
             }
-            handleGeneralError(response, status);
+            throw handleGeneralError(response, status);
         } else {
             throw new NotCancellableException();
```

Comment:
This is solved in commit https://github.com/digipost/signature-api-client-java/commit/b49a6b09a307acff615331ec31d63e25c7adbba1. Please see the commit message for a detailed explanation :)